src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -181,6 +184,33 @@ def register_options(cls, register):
       ]
     )
 
+    cls.register_jvm_tool(
+      register,
+      'scalac-outliner',
+      classpath=[
+        JarDependency(
+          org='org.scala-lang',
+          name='scala-compiler',
+          rev='2.12.10',
+        ),
+      ]
+    )
+
+    cls.register_jvm_tool(
+      register,
+      'wartremover',
+      classpath=[
+        JarDependency(
+          org='org.wartremover',
+          name='wartremover_2.12.10',
+          rev='2.4.3',
+        ),
+      ],
+      custom_rules=[
+        Shader.exclude_package('wartremover', recursive=True),
+      ]
+    )
+
```

Comment:
wartremover seems to also contain the similar dependencies such as `scala-library-2.12.10.jar`

Resolve for scalac:
```
$ ./coursier fetch -t org.scala-lang:scala-compiler:2.12.10
  Result:
└─ org.scala-lang:scala-compiler:2.12.10
   ├─ org.scala-lang:scala-library:2.12.10
   ├─ org.scala-lang:scala-reflect:2.12.10
   │  └─ org.scala-lang:scala-library:2.12.10
   └─ org.scala-lang.modules:scala-xml_2.12:1.0.6
      └─ org.scala-lang:scala-library:2.12.0 -> 2.12.10
/Users/yic/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-compiler/2.12.10/scala-compiler-2.12.10.jar
/Users/yic/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-reflect/2.12.10/scala-reflect-2.12.10.jar
/Users/yic/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/modules/scala-xml_2.12/1.0.6/scala-xml_2.12-1.0.6.jar
/Users/yic/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.12.10/scala-library-2.12.10.jar
```
Resolve for wartremover:
```
$ ./coursier fetch -t org.wartremover:wartremover_2.12.10:2.4.3
https://repo1.maven.org/maven2/org/wartremover/wartremover_2.12.10/2.4.3/wartremover_2.12.10-2.4.3.pom
  100.0% [##########] 2.0 KiB (8.1 KiB / s)
  Result:
└─ org.wartremover:wartremover_2.12.10:2.4.3
   ├─ org.scala-lang:scala-compiler:2.12.10
   │  ├─ org.scala-lang:scala-library:2.12.10
   │  ├─ org.scala-lang:scala-reflect:2.12.10
   │  │  └─ org.scala-lang:scala-library:2.12.10
   │  └─ org.scala-lang.modules:scala-xml_2.12:1.0.6
   │     └─ org.scala-lang:scala-library:2.12.0 -> 2.12.10
   └─ org.scala-lang:scala-library:2.12.10
https://repo1.maven.org/maven2/org/wartremover/wartremover_2.12.10/2.4.3/wartremover_2.12.10-2.4.3.jar
  100.0% [##########] 303.9 KiB (6.5 MiB / s)
/Users/yic/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-reflect/2.12.10/scala-reflect-2.12.10.jar
/Users/yic/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/modules/scala-xml_2.12/1.0.6/scala-xml_2.12-1.0.6.jar
/Users/yic/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-compiler/2.12.10/scala-compiler-2.12.10.jar
/Users/yic/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/scala-lang/scala-library/2.12.10/scala-library-2.12.10.jar
/Users/yic/Library/Caches/Coursier/v1/https/repo1.maven.org/maven2/org/wartremover/wartremover_2.12.10/2.4.3/wartremover_2.12.10-2.4.3.jar
```
so maybe if you put them together
```
    cls.register_jvm_tool(
      register,
      'scalac-outline',
      classpath=[
        JarDependency(
          org='org.scala-lang',
          name='scala-compiler',
          rev='2.12.10',
        ),
        JarDependency(
          org='org.wartremover',
          name='wartremover_2.12.10',
          rev='2.4.3',
        ),
      ],
      custom_rules=[
        shader_rule_for_wartremover + shader_rule_for_scalac
      ]
    )
```

Reply:
The dependencies of wartremover should not be needed for the `-Xplugin:` flag to scalac. Coursier has a --intransitive flag to skip fetching dependencies
```
cs fetch --intransitive org.wartremover:wartremover_2.12.10:2.4.3
```